lib/Frontend/CachingUtils.cpp

```diff
@@ -281,7 +297,10 @@ bool replayCachedCompilerOutputs(
                     toString(File.takeError()));
       continue;
     }
-    *File << Output.Proxy.getData();
+
+    if (!UseCASBackend)
+      *File << Output.Proxy.getData();
```

Comment:
The output here is more than just object file. I think you might have dropped all other outputs when using MCCAS. Please check those files exist on disk in the test.